che/service/che_service.go

```diff
@@ -1,30 +1,45 @@
 package service
 
 import (
-	"fmt"
 	"context"
+	"fmt"
 	"net/http"
 
+	"github.com/fabric8-services/fabric8-auth/application/service"
+	"github.com/fabric8-services/fabric8-auth/application/service/base"
+	servicecontext "github.com/fabric8-services/fabric8-auth/application/service/context"
+	"github.com/fabric8-services/fabric8-auth/authorization/token/manager"
 	"github.com/fabric8-services/fabric8-auth/log"
 	"github.com/fabric8-services/fabric8-auth/rest"
-
-	"github.com/fabric8-services/fabric8-auth/application/service"
 	errs "github.com/pkg/errors"
 )
 
 // cheServiceImpl is the default implementation of CheService.
 type cheServiceImpl struct {
-	config Configuration
+	base.BaseService
+	config       Configuration
+	tokenManager manager.TokenManager
 }
 
 // Configuration the config for the Che service
 type Configuration interface {
+	manager.TokenManagerConfiguration
 	GetCheServiceURL() string
 }
+
 // NewCheService creates a new Che service.
-func NewCheService(config Configuration) service.CheService {
+func NewCheService(context servicecontext.ServiceContext, config Configuration) service.CheService {
+	tokenManager, err := manager.NewTokenManager(config)
+	if err != nil {
+		log.Panic(nil, map[string]interface{}{
```

Comment:
why `Panic` here?

Reply:
Good catch, this was copied straight from the token service.. i'll change it to something more sensible

Reply:
I've removed the tokenManager member from che_service, and instead exposed it in TokenService so that we can access the token manager functions from anywhere that has access to services.